### PR TITLE
irrd: 4.3.0.post1 -> 4.4.2

### PR DESCRIPTION
**Pull request by yu-re-ka**
## Description of changes

irrd currently doesn't build on master and release-23.11 because of test failures. updating fixes that.

#### todo

- [x] check if new packages should be built using format = pyproject and poetry-core
- [x] make sure all new packages have tests
- [x] add meta sections

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### Priorities

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/ariadne/remove-opentracing.patch
[the remark this answers is not in the document]
Comment (yu-re-ka) [reply pkgs/development/python-modules/ariadne/remove-opentracing.patch:1]: There is an include in `tests/asgi/conftest.py` which prevents use from running any tests at all if opentracing is not installed

**Comment by yu-re-ka** [comment]
split out the the ariadne fix into #271999 for easier review

File: pkgs/servers/misc/irrd/default.nix [hunk 1]
```diff
@@ -19,31 +20,23 @@ let
         };
         doCheck = false;
       });
-      starlette = prev.starlette.overridePythonAttrs (oldAttrs: rec {
-        version = "0.20.4";
+      flask-sqlalchemy = prev.flask-sqlalchemy.overridePythonAttrs (oldAttrs: rec {
+        version = "2.5.1";
```
Comment (johannwagner) [inline pkgs/servers/misc/irrd/default.nix:24]: Would you mind adding a hint, why we do an override here? 
Probably "irrd needs an v2 version of flask-sqlachemy, we package a v3"?

File: pkgs/servers/misc/irrd/default.nix [hunk 2]
```diff
@@ -134,10 +139,16 @@ py.pkgs.buildPythonPackage rec {
       sleep 1
     done
 
+    export SMTPD_HOST=127.0.0.1
     export IRRD_DATABASE_URL="postgres:///$PGDATABASE"
     export IRRD_REDIS_URL="redis://localhost/1"
   '';
 
+  postgresqlTestSetupPost = ''
+    echo "track_commit_timestamp=on" >> $PGDATA/postgresql.conf
```
Comment (johannwagner) [inline pkgs/servers/misc/irrd/default.nix:148]: Why do we need this for our tests? Documentation only states to use this for query-only replicas? Similar to `hot_standby_feedback`?

**Review by johannwagner — APPROVED** [review]

**Comment by johannwagner** [comment]
I just have two nitpicks, looks good otherwise, no blockers here. Probably also want to backport this.

File: pkgs/servers/misc/irrd/default.nix (hunk 1 above)
Comment (yu-re-ka) [reply pkgs/servers/misc/irrd/default.nix:24]: Whoops that is not actually required

File: pkgs/servers/misc/irrd/default.nix (hunk 2 above)
Comment (yu-re-ka) [reply pkgs/servers/misc/irrd/default.nix:148]: Log when building without this option: https://termbin.com/hosb

Comment (johannwagner) [reply pkgs/servers/misc/irrd/default.nix:137]: Oh, I missed the comment above, makes sense.

**Review by johannwagner — APPROVED** [review]

File: pkgs/development/python-modules/starlette-wtf/default.nix [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+, python-multipart
+, starlette
+, wtforms
+, pytestCheckHook
+}:
+
+buildPythonPackage rec {
+  pname = "starlette-wtf";
+  version = "0.4.3";
+  format = "setuptools";
```
Comment (mweinelt) [inline pkgs/development/python-modules/starlette-wtf/default.nix:14]: Use pyproject format and pass setuptools via `nativeBuildInputs`

**Review by mweinelt — DISMISSED** [review]

**Comment by yu-re-ka** [comment]
Result of `nixpkgs-review pr 271179` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>22 packages built:</summary>
  <ul>
    <li>irrd</li>
    <li>irrd.dist</li>
    <li>python310Packages.asgi-logger</li>
    <li>python310Packages.asgi-logger.dist</li>
    <li>python310Packages.imia</li>
    <li>python310Packages.imia.dist</li>
    <li>python310Packages.smtpdfix</li>
    <li>python310Packages.smtpdfix.dist</li>
    <li>python310Packages.starlette-wtf</li>
    <li>python310Packages.starlette-wtf.dist</li>
    <li>python310Packages.wtforms-bootstrap5</li>
    <li>python310Packages.wtforms-bootstrap5.dist</li>
    <li>python311Packages.asgi-logger</li>
    <li>python311Packages.asgi-logger.dist</li>
    <li>python311Packages.imia</li>
    <li>python311Packages.imia.dist</li>
    <li>python311Packages.smtpdfix</li>
    <li>python311Packages.smtpdfix.dist</li>
    <li>python311Packages.starlette-wtf</li>
    <li>python311Packages.starlette-wtf.dist</li>
    <li>python311Packages.wtforms-bootstrap5</li>
    <li>python311Packages.wtforms-bootstrap5.dist</li>
  </ul>
</details>

**Pull request merged**